R/Data-validity.R

```diff
@@ -181,6 +181,67 @@ v_data_da <- function(object) {
   v$result()
 }
 
+#' @describeIn v_data_objects validates that the [`DataOrdinal`] object
+#' contains valid elements with respect to their types, dependency and length.
+v_data_ordinal <- function(object) {
+  v <- Validate()
+  v$check(v_general_data(object))
```

Comment:
Yes, there's lots of duplication because of the decision to make `DataOrdinal` a sibling of `Data` rather than a child.  Since we did this to minimise changes to `Data`, I decided not to create these helper functions - because that would mean changing `Data`.  But I agree - using helpers is the better option.  Done.